### PR TITLE
Integration Mediator

### PR DESCRIPTION
**Pull request by nityan**

### CODE REVIEW

File: src/main/resources/mediator-registration-info.json [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
-  "description": "An openHIM mediator for handling integration between Health Facility Registry and various health systems",
+  "name": "HFR Mediator",
+  "description": "A mediator for handling sharing of health facilities information from Health Facility Registry to other health information systems",
   "endpoints": [
     {
-      "name": "HFR-Mediator Route",
+      "name": "HFR Mediator Route",
       "host": "localhost",
-      "port": "3003",
+      "port": "3001",
```
Comment (cozej4) [inline src/main/resources/mediator-registration-info.json]: Please update this to 3003, Port 3001 is already taken by another mediator

File: src/main/resources/mediator.properties [hunk 2]
```diff
@@ -1,11 +1,23 @@
 # Mediator Properties
 mediator.name=HFR-Mediator
 mediator.host=localhost
-mediator.port=3003
+mediator.port=3001
```
Comment (cozej4) [inline src/main/resources/mediator.properties]: Please update this to 3003, Port 3001 is already taken by another mediator

File: src/test/java/tz/go/moh/him/hfr/mediator/mock/MockDestination.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+        HfrRequest request = gson.fromJson(msg.getBody(), HfrRequest.class);
+
+//        assertEquals(expectedPayload.getPartNum(), dailyStockStatus.getPartNum());
+//        assertEquals(expectedPayload.getDate(), dailyStockStatus.getDate());
+//        assertEquals(expectedPayload.getMonthOfStock(), dailyStockStatus.getMonthOfStock());
+//        assertEquals(expectedPayload.getOnHandQty(), dailyStockStatus.getOnHandQty());
+//        assertEquals(expectedPayload.getOum(), dailyStockStatus.getOum());
+//        assertEquals(expectedPayload.getPartDescription(), dailyStockStatus.getPartDescription());
+//        assertEquals(expectedPayload.getPlant(), dailyStockStatus.getPlant());
```
Comment (cozej4) [inline src/test/java/tz/go/moh/him/hfr/mediator/mock/MockDestination.java]: I would suggest you assert on that what was sent by the mediator is what is being received by the mock destination system.

**Review by cozej4 — CHANGES_REQUESTED** [review]

File: src/test/java/tz/go/moh/him/hfr/mediator/mock/MockDestination.java (hunk 3 above)
Comment (cozej4) [reply src/test/java/tz/go/moh/him/hfr/mediator/mock/MockDestination.java]: You might will also need to add additional unit tests to increase the coverage to at least 60% to be allowed to merge the pull request.

File: src/main/resources/mediator.properties (hunk 2 above)
Comment (nityan) [reply src/main/resources/mediator.properties]: Changes are up

File: src/main/resources/mediator-registration-info.json (hunk 1 above)
Comment (nityan) [reply src/main/resources/mediator-registration-info.json]: Changes are up

File: src/test/java/tz/go/moh/him/hfr/mediator/mock/MockDestination.java (hunk 3 above)
Comment (nityan) [reply src/test/java/tz/go/moh/him/hfr/mediator/mock/MockDestination.java]: @cozej4 I've updated the tests now, the internal IntelliJ coverage report states 74% after my latest commit, so hopefully it's good now :)

File: src/test/java/tz/go/moh/him/hfr/mediator/mock/MockOpenHIM.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+            MediatorHTTPResponse mediatorHTTPResponse = new MediatorHTTPResponse((MediatorHTTPRequest) msg, openHIMSampleTransaction, 200, headers);
+            workingRequest.getRespondTo().tell(mediatorHTTPResponse, getSelf());
+        } else if (workingRequest.getMethod().equals("PUT")) {
+            JSONObject receivedUpdatedTransaction = new JSONObject(((MediatorHTTPRequest) msg).getBody());
+
+            Assert.assertEquals(200, receivedUpdatedTransaction.getJSONObject("response").getInt("status"));
+
+            Assert.assertEquals("Successful", receivedUpdatedTransaction.getString("status"));
+//                Assert.assertEquals(jsonElmisAckPayload, receivedUpdatedTransaction.getJSONObject("response").getString("body"));
```
Comment (cozej4) [inline src/test/java/tz/go/moh/him/hfr/mediator/mock/MockOpenHIM.java]: Clean this unused code

**Review by cozej4 — CHANGES_REQUESTED** [review]

File: src/test/java/tz/go/moh/him/hfr/mediator/mock/MockOpenHIM.java (hunk 4 above)
Comment (nityan) [reply src/test/java/tz/go/moh/him/hfr/mediator/mock/MockOpenHIM.java]: Will do

**Review by cozej4 — DISMISSED** [review]

**Review by mo3dev — APPROVED** [review]
good

**Pull request merged**